kkv 2.0

NiklasJonsson6 opened the pull request:
No more sidecars!

NiklasJonssonYolean committed:
new quarkus version, updated dependencies and build with jib

NiklasJonssonYolean committed:
A config property cannot (anymore?) be converted to a Duration by default, so lets just keep the string

NiklasJonssonYolean committed:
configuration for new features

NiklasJonssonYolean committed:
a bean that watches the target service endpoints configured and keeps a list of targets and their addresses

NiklasJonssonYolean committed:
a new UpdatesDispatcher using vertx-webclient and targets from EndpointsWatcher rather than a statically configured set of targets

NiklasJonssonYolean committed:
OnUpdateForwarder now uses a single injected UpdatesDispatcher to dispatch updates, and lets the dispatcher handle targets instead of keeping a configured set of targets, and a dispatcher for each target

NiklasJonssonYolean committed:
lets keep the artifact id but call this version 2.0, and quarkus 2.16

NiklasJonssonYolean committed:
document depcrecated test deploy (skaffold build is maintained)

NiklasJonssonYolean committed:
sort out dispatcher config

NiklasJonssonYolean committed:
logging fix

NiklasJonssonYolean committed:
we still call this service kafka-keyvalue

solsson commented on src/main/java/se/yolean/kafka/keyvalue/ConsumerAtLeastOnce.java:
```diff
@@ -67,8 +67,8 @@ public enum Stage {
 
-  @ConfigProperty(name = "kkc.assignments.timeout", defaultValue="90s")
-  private Duration assignmentsTimeout;
+  @ConfigProperty(name = "kkc.assignments.timeout", defaultValue="90")
```
486be0b40ed38bdaaeb13df3c3143fe653e35c5b is breaking if we have used the config anywhere. I think it would be worthwhile resurrecting Duration parsing.

solsson commented on skaffold.yaml:
```diff
@@ -9,23 +9,36 @@ build:
   - linux/arm64/v8
```
When we update skaffold.yaml we should change the first line to apiVersion `skaffold/v3`

solsson commented:
Backwards compatibility is out of scope for this PR, see https://github.com/Yolean/kafka-keyvalue/commit/7a91c5c782680352e4d8b75c6b1cd4610a5d8e63, but I'd still like to see Duration values work because strings are more convenient than integers in yaml (k8s pod spec env or args).

I approve of the env renames to upper case. They were lowercase for some legacy reason that I've forgotten about.

Given the limited test coverage of KKV we need to run in QA from this PR. It would be interesting to have KKV as a case for trying out [skaffold verify](https://skaffold.dev/docs/pipeline-stages/verify/) in ephemeral namespaces. A redpanda container and a "client" deployment with metrics could do wonders.

NiklasJonssonYolean committed:
skaffold apiVersion

NiklasJonssonYolean committed:
use the Duration type for the timeout config proeprty

again, it works but the microprofile language server does not seem to understand automatic converters

solsson commented:
Pushed followup commits to a branch in our repo. Saves time.

https://github.com/Yolean/kafka-keyvalue/compare/kkv2?expand=1
